src/ir_Kelon.cpp

```diff
@@ -0,0 +1,486 @@
+// Copyright 2021 Davide Depau
+
+/// @file
+/// @brief Support for Kelan AC protocol.
+/// Both sending and decoding should be functional for models of series KELON ON/OFF 9000-12000.
+/// All features of the standard remote are implemented.
+///
+/// @note Unsupported:
+///    - Explicit on/off due to AC unit limitations
+///    - Explicit swing position due to AC unit limitations
+///    - Fahrenheit.
+
+#include <cstring>
+
+#include "ir_Kelon.h"
+
+#include "IRrecv.h"
+#include "IRsend.h"
+#include "IRutils.h"
+#include "IRtext.h"
+
+
+using irutils::addBoolToString;
+using irutils::addIntToString;
+using irutils::addSignedIntToString;
+using irutils::addModeToString;
+using irutils::addFanToString;
+using irutils::addTempToString;
+using irutils::addLabeledString;
+using irutils::minsToString;
+
+// Constants
+const uint16_t kKelonHdrMark = 9000;
+const uint16_t kKelonHdrSpace = 4600;
+const uint16_t kKelonBitMark = 560;
+const uint16_t kKelonOneSpace = 1680;
+const uint16_t kKelonZeroSpace = 600;
+const uint32_t kKelonGap = kDefaultMessageGap;
+const uint16_t kKelonFreq = 38000;
+
+#if SEND_KELON
+
+/// Send a Kelon message.
+/// Status: Beta / Should be working.
```

Comment:
If it is working for you, you should list it as STABLE.

Reply:
I'll keep these open and do some final testing with the actual AC before this is merged.